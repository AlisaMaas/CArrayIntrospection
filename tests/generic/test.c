#include "structHeader.h"
int find(struct foo *f, char dest) {
    for (int i = 0; f->string[i]; i++) {
        if (dest == f->string[i]) {
            return 1;
        }
    }
    return 0;
}

void setString(struct foo *f, char* string) {
    f->string = string;
}