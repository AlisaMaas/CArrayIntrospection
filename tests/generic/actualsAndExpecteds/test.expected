setString with argument 1 should be annotated NULL_TERMINATED (Sentinel-terminated by 0)  because  pushed information from a store to string1 from string
struct struct.foo[0] should be annotated NULL_TERMINATED (Sentinel-terminated by 0) because found a non optional sentinel check in find.
